operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java

```diff
@@ -67,7 +67,7 @@ private void init(LeaderElectionConfiguration config) {
                 config.getRenewDeadline(),
                 config.getRetryPeriod(),
                 leaderCallbacks(),
-                true,
+                false, // this is required to receive stop event in all cases
```

Comment:
Why, though?

Reply:
what do you mean? we want to get those all cases

Reply:
What I mean is that the comment doesn't explain why changing the value to `false` makes sure that `stop` is called…

Reply:
That should be part of the fabric8 client docs rather, actually not there but it's in the go docs?
https://pkg.go.dev/k8s.io/client-go/tools/leaderelection#LeaderElectionConfig
Should I link this there?

Reply:
Maybe but mostly explain why this "release on cancel" setting has an impact on the lifecycle of our manager.

Reply:
extended the comment, but basically this should be very straghforward, that the onStop is called, also if the lock is released.